sqlx-mysql/src/query_result.rs

```diff
@@ -24,3 +24,14 @@ impl Extend<MySqlQueryResult> for MySqlQueryResult {
         }
     }
 }
+
+#[cfg(feature = "any")]
+/// This conversion lose the last insert id data.
+impl From<MySqlQueryResult> for crate::any::AnyQueryResult {
+    fn from(done: MySqlQueryResult) -> Self {
+        crate::any::AnyQueryResult {
+            rows_affected: done.rows_affected(),
+            last_insert_id: None,
```

Comment:
Instead of throwing away `last_insert_id`, use `.try_into().ok()` to convert to `i64` if not out-of-bounds.

Reply:
[rewrite use and address implementation concerns](https://github.com/launchbadge/sqlx/pull/3608/commits/b5c279abedf0a82741b7f1802e17479ca2a64f3b)